### ISSUE TITLE
Forge 47.1.0 compatibility

### ISSUE THREAD

**Issue by zekeflynn** [Skidamek/AutoModpack#171]
### Motivation

It's the recommended forge version.

### Description

Hi is it possible to add compatibility to the 47.1.0 version of forge?

**Comment by Skidamek** [comment]
Yes, for the time being you can use this build.
[automodpack-mc1.20.1-forge-3.5.3-SHAPSHOT.zip](https://github.com/Skidamek/AutoModpack/files/12652320/automodpack-mc1.20.1-forge-3.5.3-SHAPSHOT.zip)

**Comment by zekeflynn** [comment]
Oh wow thank you! 😊

**Comment by zekeflynn** [comment]
PS: Didn't know I could close it, didn't mean to, sorry for troubles.

**Comment by zekeflynn** [comment]
> [13:19:56] [Server thread/INFO] [minecraft/ServerLoginPacketListenerImpl]: com.mojang.authlib.GameProfile@6cedd80c[id=3a4b7746-070d-47f9-8c89-adc72538defc,name=Rob_Flynn,properties={textures=[com.mojang.authlib.properties.Property@4cc55510]},legacy=false] (/179.213.160.37:57417) lost connection: Disconnected

Do you happen to know why I get this when trying to connect? It downloaded all the mods correctly to my automodpack folder in the client but still won't let me connect to the server, whenever I boot MC the mods from the automodpack folder don't seem to load in as the default mods folder remains empty.

**Comment by Skidamek** [comment]
Try deleting automodpack folder and redownload modpack from game again

**Comment by zekeflynn** [comment]
I launch and connect to the server and it counts up the number of URL's to download and after it finishes it seems to sit on this screen forever 😥
![image](https://github.com/Skidamek/AutoModpack/assets/33818126/f1ae8a6a-2261-47a9-824f-62c28209953b)

The server console prints the same message as before:
>[00:45:46] [Server thread/INFO] [minecraft/ServerLoginPacketListenerImpl]: com.mojang.authlib.GameProfile@be4219[id=3a4b7746-070d-47f9-8c89-adc72538defc,name=Rob_Flynn,properties={textures=[com.mojang.authlib.properties.Property@26dec1d8]},legacy=false] (/179.213.160.37:62355) lost connection: Disconnected

**Comment by Skidamek** [comment]
Could you send client log?

**Comment by zekeflynn** [comment]
[latest.log](https://github.com/Skidamek/AutoModpack/files/12689949/latest.log)
[debug.log](https://github.com/Skidamek/AutoModpack/files/12689950/debug.log)

Don't see anything 😔

**Comment by zekeflynn** [comment]
[latest.log](https://github.com/Skidamek/AutoModpack/files/12690168/latest.log)


another one where it's crashing the client on boot

**Comment by zekeflynn** [comment]
Specifically
>[22Sep2023 11:04:28.990] [main/ERROR] [AutoModpack/]: Error while initializing modpack updater

**Comment by Skidamek** [comment]
Does it also happens on forge 47.1.3?

**Comment by zekeflynn** [comment]
I gave in and updated everything to 47.1.3 even though this started with me asking for 47.1.0 compatibility 😆

But if it means I can get it to work or help fixing something I'm up for it.

Sadly, still get the same error on 47.1.3 lol

```
[23Sep2023 12:51:02.170] [main/INFO] [cpw.mods.modlauncher.Launcher/MODLAUNCHER]: ModLauncher running: args [--username, Rob_Flynn, --version, 1.20.1-forge-47.1.30, --gameDir, C:\Users\Flynn\AppData\Roaming\.minecraft\.modtest, --assetsDir, C:\Users\Flynn\AppData\Roaming\.minecraft\assets, --assetIndex, 5, --uuid, 3a4b7746070d47f98c89adc72538defc, --accessToken, ????????, --clientId, OTA4MzhmNzAtZmUyNi00MTJjLWI5YTYtNTFmNWRlN2Y0NzUw, --xuid, 2535453252197787, --userType, msa, --versionType, release, --width, 2500, --height, 1200, --quickPlayPath, C:\Users\Flynn\AppData\Roaming\.minecraft\quickPlay\java\1695484259728.json, --launchTarget, forgeclient, --fml.forgeVersion, 47.1.30, --fml.mcVersion, 1.20.1, --fml.forgeGroup, net.minecraftforge, --fml.mcpVersion, 20230612.114412]
[23Sep2023 12:51:02.174] [main/INFO] [cpw.mods.modlauncher.Launcher/MODLAUNCHER]: ModLauncher 10.0.9+10.0.9+main.dcd20f30 starting: java version 17.0.8 by Microsoft; OS Windows 10 arch amd64 version 10.0
[23Sep2023 12:51:03.585] [main/INFO] [net.minecraftforge.fml.loading.ImmediateWindowHandler/]: Loading ImmediateWindowProvider fmlearlywindow
[23Sep2023 12:51:03.688] [main/INFO] [EARLYDISPLAY/]: Trying GL version 4.6
[23Sep2023 12:51:03.798] [main/INFO] [EARLYDISPLAY/]: Requested GL version 4.6 got version 4.6
[23Sep2023 12:51:03.896] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=union:/C:/Users/Flynn/AppData/Roaming/.minecraft/libraries/org/spongepowered/mixin/0.8.5/mixin-0.8.5.jar%23100!/ Service=ModLauncher Env=CLIENT
[23Sep2023 12:51:03.947] [pool-2-thread-1/INFO] [EARLYDISPLAY/]: GL info: NVIDIA GeForce GTX 1660/PCIe/SSE2 GL version 4.6.0 NVIDIA 537.13, NVIDIA Corporation
[23Sep2023 12:51:03.958] [main/INFO] [APTweaks(Mods)/]: ? Processing 120 client sides mods with 0 skipped entries ...
[23Sep2023 12:51:03.992] [main/INFO] [APTweaks(Mods)/]: ? No need to update client side mods config file C:\Users\Flynn\AppData\Roaming\.minecraft\.modtest\config\adaptive_performance_tweaks\client_side_mods.list
[23Sep2023 12:51:03.993] [main/INFO] [APTweaks(Mods)/]: ? Processing 120 client sides mods with 3 skipped entries ...
[23Sep2023 12:51:03.999] [main/INFO] [APTweaks(Mods)/]: [Mod Optimizer] ? init with game dir C:\Users\Flynn\AppData\Roaming\.minecraft\.modtest and mods dir C:\Users\Flynn\AppData\Roaming\.minecraft\.modtest\mods for target CLIENT
[23Sep2023 12:51:04.000] [main/INFO] [APTweaks(Mods)/]: [Mod Optimizer] ? Optimizing Duplicated Mods ...
[23Sep2023 12:51:04.002] [main/INFO] [APTweaks(Mods)/]: [Duplicated Mods] checking ~246 mods in C:\Users\Flynn\AppData\Roaming\.minecraft\.modtest\mods for duplication with file extension .jar ...
[23Sep2023 12:51:04.820] [main/INFO] [APTweaks(Mods)/]: [Mod Optimizer] ? Re-Enable possible client side mods ...
[23Sep2023 12:51:05.674] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file C:\Users\Flynn\AppData\Roaming\.minecraft\libraries\net\minecraftforge\fmlcore\1.20.1-47.1.30\fmlcore-1.20.1-47.1.30.jar is missing mods.toml file
[23Sep2023 12:51:05.677] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file C:\Users\Flynn\AppData\Roaming\.minecraft\libraries\net\minecraftforge\javafmllanguage\1.20.1-47.1.30\javafmllanguage-1.20.1-47.1.30.jar is missing mods.toml file
[23Sep2023 12:51:05.679] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file C:\Users\Flynn\AppData\Roaming\.minecraft\libraries\net\minecraftforge\lowcodelanguage\1.20.1-47.1.30\lowcodelanguage-1.20.1-47.1.30.jar is missing mods.toml file
[23Sep2023 12:51:05.682] [main/WARN] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Mod file C:\Users\Flynn\AppData\Roaming\.minecraft\libraries\net\minecraftforge\mclanguage\1.20.1-47.1.30\mclanguage-1.20.1-47.1.30.jar is missing mods.toml file
[23Sep2023 12:51:05.953] [main/WARN] [net.minecraftforge.jarjar.selection.JarSelector/]: Attempted to select two dependency jars from JarJar which have the same identification: Mod File:  and Mod File: . Using Mod File: 
[23Sep2023 12:51:05.953] [main/INFO] [net.minecraftforge.fml.loading.moddiscovery.JarInJarDependencyLocator/]: Found 33 dependencies adding them to mods collection
[23Sep2023 12:51:06.050] [main/INFO] [PreloadingTricks/LanguageProvider/]: [PreloadingTricks/LanguageProvider] Loading settingdust.preloadingtricks.fabric.FabricLanguageProviderCallback
[23Sep2023 12:51:06.055] [main/INFO] [PreloadingTricks/LanguageProvider/]: [PreloadingTricks/LanguageProvider] Loading settingdust.preloadingtricks.quilt.QuiltLanguageProviderCallback
[23Sep2023 12:51:06.067] [main/INFO] [PreloadingTricks/LanguageProvider/]: [PreloadingTricks/LanguageProvider] Loading settingdust.preloadingtricks.forge.ForgeLanguageProviderCallback
[23Sep2023 12:51:06.073] [main/INFO] [PreloadingTricks/ModSetup/]: [PreloadingTricks/ModSetup] Loading pl.skidam.automodpack_core.Preload
[23Sep2023 12:51:06.075] [main/INFO] [AutoModpack/]: Prelaunching AutoModpack...
[23Sep2023 12:51:06.097] [main/INFO] [AutoModpack/]: Loaded config! took 18ms
[23Sep2023 12:51:07.268] [main/INFO] [AutoModpack/]: Checking if AutoModpack is up-to-date...
[23Sep2023 12:51:07.754] [main/WARN] [AutoModpack/]: Couldn't get latest version of AutoModpack from Modrinth API. Likely automodpack isn't updated to your version of minecraft yet...
[23Sep2023 12:51:07.773] [main/INFO] [AutoModpack/]: Modpack hash is the same as server modpack hash
[23Sep2023 12:51:07.774] [main/INFO] [AutoModpack/]: Modpack is up to date
[23Sep2023 12:51:08.786] [main/ERROR] [AutoModpack/]: Error while initializing modpack updater
[23Sep2023 12:51:09.020] [main/INFO] [AutoModpack/]: AutoModpack prelaunched! took 2946ms

```

**Comment by zekeflynn** [comment]
Any idea of what else I could try? 🙂

**Comment by Skidamek** [comment]
Fixed in 4.0.0 beta 1!
